Updated Mac installation instructions (#107)
UBC-MDS/UBC-MDS.github.io: resources_pages/install_ds_stack_mac.md

git version 2.24.2 (Apple Git-127)
```

> **Note:** If you run into trouble, please see that Install Git > Mac OS section from [Happy Git and GitHub for the useR](http://happygitwithr.com/install-git.html#mac-os) for additional help or strategies for Git installation.

### Configuring Git user info

Next, we need to configure Git by telling it your name and email. To do this type the following into the terminal (replacing Jane Doe and janedoe@example.com, with your name and email (the same used to sign up for GitHub), respectively):

```bash
git config --global user.name "Jane Doe"
git config --global user.email janedoe@example.com
```

> **Note:** to ensure that you haven't made a typo in any of the above, you can view your global Git configurations by either opening the configuration file in a text editor (e.g. via the command `code ~/.gitconfig`) or by typing `git config --list --global`.

### Setting VS Code as the default editor

> To make programs run from the terminal (such as `git`) use vscode by default, we will modify `~/.bash_profile`. First, maopen it using VS Code:

Reviewer: ```suggestion
To make programs run from the terminal (such as `git`) use vscode by default, we will modify `~/.bash_profile`. First, open it using VS Code (this will also create the file if it doesn't already exist):
```

Author: Thanks, yes I think we can delete this note box.

```bash
code ~/.bash_profile
```

> **Note**: if the file `~/.bash_profile` does not already exist, make one using the following command:
>
> ```bash

> > touch ~/.bash_profile

Reviewer: The file should automatically be created when you run the `code` command above. Could you try with a random non-existing file name and confirm that the file is created when you save in `vscode` without having to `touch` it first??

Author: You're right. `code ~/.bash_profile` creates the file if it doesn't exist, but only if we save it right there. I think it's still helpful to remind the students that the file might not be already there.

Reviewer: Good idea, I added a parenthesized suggestion above. Do you think that is enough and we can delete this section?

Author: Thanks, yes I think we can delete this note box.

> > **Note**: if the file `~/.bash_profile` does not already exist, make one using the following command:
> >
> > ```bash
> > touch ~/.bash_profile
> > ```
> 

Reviewer: ```suggestion
```

we will add the more up to date  channel,
To add the conda-forge channel by typing the following in the terminal:

```bash
conda config --add channels conda-forge
```

To install packages individually, we can now use the following command: `conda install <package-name>`. Let's install the key packages needed for the start of our program:

```bash
conda install \
 jupyterlab=3.* \
 numpy=1.* \
 pandas=1.* \
 flake8=3.* \
 black=21.*
```

`conda` will show you the packages that will be downloaded,
and you can press enter to proceed with the installation.
If you want to answer `yes` by default and skip this confirmation step,
you can replace `conda install` with `conda install -y`.

> **Note:** we will use many more packages than those listed above across the MDS program, however we will manage these using virtual environments (which you will learn about in DSCI 521: Platforms for Data Science).

## JupyterLab setup

We will be using the Jupytext Python package and the JupyterLab git extension to facilitate using Jupyter notebooks with Git & GitHub. Install them via the following commands:

```bash
conda install nodejs=15.*

> conda install -c conda-forge jupyterlab jupyterlab-git jupytext

Reviewer: Note to self to make this one line or remove the nodejs part if we go with another solution for the keyboard text shortcuts.

We will install the lightest possible version of LaTeX and it's necessary packages as possible so that we can render Jupyter notebooks and R Markdown documents to html and PDF. If you have previously installed LaTeX, please uninstall it before proceeding with these instructions.

First, run the following command to make sure that `/usr/local/bin` is writable:

>```shell
> sudo chown -R $(whoami):admin /usr/local/bin
>```

> **Note:** You might be asked to enter your password during installation.

Now open RStudio and run the following commands to install the `tinytex` package and setup `tinytex`:

```R
install.packages('tinytex')
tinytex::install_tinytex()
```

```bash

> ~/Library/TinyTeX/bin/*/tlmgr path add

Reviewer: In the old instructions we recommend running this right after the `chown` command. Is that not needed? Did you need to run this at all?

Author: No! I didn't need it. What is the purpose of this command?

Reviewer: It seems to be mostly for documentation related things https://www.tug.org/texlive/doc/tlmgr.html#path. We might be able to remove it since it shouldn't be needed when using Rmarkdown (although maybe it gives issues with Jupyter if we don't have it, but that seems to be fine for you) https://yihui.org/tinytex/#maintenance.

Author: So let's keep it for now?

Reviewer: Yeah, let's just keep it and if I have time I will explore and remove it (probably wont)

## PostgreSQL

We will be using PostgreSQL as our database management system. You can [download PostgreSQL 12.4 from [here](https://www.enterprisedb.com/downloads/postgresql) (do *not* select version 13). Follow the instructions for the installation. In the password page, type whatever password you want, **but make sure you'll remember it later**. For all the other options, use the default. You do not need to run "StackBuilder" at the end of the installation (if you accidentally launch the StackBuilder, click "cancel", you don't need to check any boxes).

> To test if the installation was successful open the `SQL Shell` app from the LaunchPad or applications directory. You will be asked to setup your configuration:

Reviewer: ```suggestion
To test if the installation was successful open the `SQL Shell` app from the LaunchPad or applications directory. You will be asked to setup your configuration:

```

Checking program and package versions...

## Operating system
ProductName:	macOS
ProductVersion:	11.4
BuildVersion:	20F71

> MISSING You need macOS Catalina (10.15.x).

Reviewer: Note to self to fix this in the script. For the the versions below, I will keep some as missing just to show students what it looks like; i'll change that manually later.

Reviewer 2: Let's scope this PR to not include the script, and make a new PR for the script.
